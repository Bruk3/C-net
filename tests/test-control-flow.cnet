if (5 < 15 || 4 > 5) { 
  return true; 
} else { 
     return false; 
} 

int x = 5;
while (x < 15) {
  x += 5; 
}

for (int i =16; x < i; ) {x -= 1;}