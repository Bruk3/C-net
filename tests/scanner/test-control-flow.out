IF
LPAREN
INTLIT(5)
LT
INTLIT(15)
OR
INTLIT(4)
GT
INTLIT(5)
RPAREN
LBRACE
RETURN
ID(true)
SEMI
RBRACE
ELSE
LBRACE
RETURN
ID(false)
SEMI
RBRACE
INT
ID(x)
ASSIGN
INTLIT(5)
SEMI
WHILE
LPAREN
ID(x)
LT
INTLIT(15)
RPAREN
LBRACE
ID(x)
PLUSEQ
INTLIT(5)
SEMI
RBRACE
FOR
LPAREN
INT
ID(i)
ASSIGN
INTLIT(16)
SEMI
ID(x)
LT
ID(i)
SEMI
RPAREN
LBRACE
ID(x)
MINUSEQ
INTLIT(1)
SEMI
RBRACE
EOF